Library/Homebrew/requirements/arch_requirement.rb

```diff
@@ -19,16 +19,34 @@ def initialize(tags)
   end
 
   satisfy(build_env: false) do
-    case @arch
+    if @arch.is_a? Array
```

Comment:
This will only work if you do `depends_on [[:x86_64]]`.

What I'd suggest is pulling some logic into `def initialize` which looks to see if each tag looks like a valid arch and remove it from the `tags` array if so and add to a `@archs` array which is then validated.

Reply:
I tried implementing that in the new commit - PTAL, I'm just now unsure why my tests didn't catch this.

Is there a better way to write the test, or would that have to be a test elsewhere outside the context of this class?

Reply:
Could have a dedicated test that passes in `:build` and ensures that it works like how you'd expect.

Reply:
I'm not sure I follow here - can you be more specific about what you mean by `:build` here?

Reply:
Add a test for `depends_on arch: [:x86_64, :build]` to ensure that it `Requirement#build?` still behaves as expected.